symfony/framework-bundle/3.3/public/index.php

```diff
@@ -23,7 +23,7 @@
 
 // Request::setTrustedProxies(['0.0.0.0/0'], Request::HEADER_FORWARDED);
 
-$kernel = new Kernel($_SERVER['APP_ENV'] ?? 'dev', $_SERVER['APP_DEBUG'] ?? true);
+$kernel = new Kernel($_SERVER['APP_ENV'] ?? 'dev', $_SERVER['APP_DEBUG'] ?? false);
```

Comment:
Setting `APP_DEBUG` to `'0'` explicitly (which triggers non-debug mode in the console) will pass  a non-empty string here, and so be truthy.

And you also need to update line 15

Reply:
L15 updated

Reply:
Anyway `(bool) '0'` should be `false` in https://github.com/symfony/symfony/blob/edb8c33f35b981b72adbe23798753703edfe3819/src/Symfony/Component/HttpKernel/Kernel.php#L83

Reply:
The bool cast should be added here. Otherwise it's not passing the right type, esp. if we add type hints for sf 4.

Reply:
@Tobion we don't have strict type enabled so this is not required: php will do the cast for us based on the type hint.